enhancements/olm/simplify-olm-apis.md

```diff
@@ -0,0 +1,366 @@
+- restrict the `ServiceAccounts` a user can install operators with
+- restrict the bundle indexes a given user can use to resolve operator bundles
+
+#### As an __Operator Installer__, I want to
+
+- view the state of an operator by inspecting the status of a single API resource
+- deploy an operator by applying its manifests directly
+- deploy an operator using a bundle image
+- deploy and upgrade an operator by referencing a bundle index
+
+#### As an __Operator User__, I want to
+
+- view the operators I can use on a cluster
+
+### Implementation Details/Notes/Constraints
+
+#### Viewing On-cluster Operators
+
+To enable viewing operators on a cluster:
+
+- Add a new cluster-scoped resource that represents a cluster
+  - call this resource `Operator`
+- On the `Operator` resource, surface:
+  - references to the resources that compose an operator
+  - operator metadata
+
+__Details:__
+
+Add the `operators.coreos.com/v2alpha1` API group version to OLM
+
+- alpha versions make no [support guarantees](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions), so resources can be iterated quickly
+
+Add a command option to `olm-operator` that enables `v1alpha2`
+
+- leave disabled by default
+
+```bash
+olm-operator --enable-v2alpha1
+```
+
+Add the cluster-scoped `Operator` resource to the `v2alpha1`
+
+- cluster-scoped resources can be viewed without granting namespace list
+- they also require a `ClusterRole` to configure any access
+
+Define a label convention for selecting resources that compose an operator
+
+- `operators.coreos.com/v2/operators/<name>: ""`; where `name` is `metadata.name`
+- using a unique label key helps avoid collisions
+- using a deterministic label key lets users know the exact label used in advance
+
+Define, as part of the `Operator` resource, a `status.resources` field that surfaces:
+
+- `matchLabels`: label selector used to list operator components
+- `components`: references to resources selected by `matchLabels`
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- `matchLabels` is resolved from the label convention and stored in the `status.resources.matchLabels` field
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operators/plumbus: ""
+```
+
+Add OLM controller logic such that, for each `Operator` resource:
+
+- cluster-scoped resources are selected using `status.resources.matchLabels`
+- namespace-scoped resources are selected __across all namespaces__ using `status.resources.matchLabels`
+- the union of both selections is stored in `status.resources.components`
+- kinds selected in both cases may be predefined or queried from discovery
+
+```yaml
+status:
+  resources:
+    matchLabels:
+      operators.coreos.com/v2/operator-name: plumbus
+    components:
+    - kind: Subscription
+      namespace: my-ns
+      name: plumbus
+      uid: d655a13e-d06a-11e9-821f-9a2e3b9d8156
```

Comment:
Is the resourceVersion useful here? 
What about the UID?

If we assume that a lot of operators _will_ deploy into one namespace, can we factor out the common namespace so we don't repeat it everywhere? (and only add it in if they span multiple namespaces?)

Reply:
I was thinking we were just going to embed the `ObjectReference` type and only fill in the extra fields when we want to provide more info. Do you we should implement the fields ourselves as we need them instead of embedding?

Reply:
We can embed the type, but I believe most fields are optional, right?

Reply:
https://godoc.org/k8s.io/api/core/v1#ObjectReference